devdoc/constbuffer_requirements.md

```diff
@@ -36,11 +36,11 @@ MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithMoveMemory, unsign
 
 MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_CreateWithCustomFree, const unsigned char*, source, size_t, size, CONSTBUFFER_CUSTOM_FREE_FUNC, customFreeFunc, void*, customFreeFuncContext);
 
-MOCKABLE_FUNCTION(, CONSTBUFFER_HANDLE, CONSTBUFFER_Clone, CONSTBUFFER_HANDLE, constbufferHandle);
```

Comment:
Why would you remove the clone API, this will require changes to the amqp transport that I'm sure people don't want to go through.  It would seem that you can keep the clone function  and also add the other IncRef and DecRef functions.

Reply:
Because Clone does not copy the contents. Semantically this is inc_ref. And since it is inc_ref then why not have it as such? Having clone that actually does inc_ref forces writing of code that checks the handle returned by Clone, but in reality that should never be checked, nor should it fail when Clone is called with a non-NULL handle.

The changes in the users are trivial (as seen in const buffer array which is a user of it).

---
In reply to: [251143791](https://github.com/Azure/azure-c-shared-utility/pull/262#discussion_r251143791) [](ancestors = 251143791)